src/Sulu/Component/Content/Import/Webspace.php

```diff
@@ -0,0 +1,511 @@
+            $this->documentManager->publish($document, $locale);
+            $this->documentManager->flush();
+            $this->documentRegistry->clear(); // FIXME else it failed on multiple page import
+
+            return true;
+        } catch (\Exception $e) {
+            $this->logger->error(
+                sprintf(
+                    '<info>%s</info>%s: <error>%s</error>%s',
+                    $uuid,
+                    PHP_EOL . get_class($e),
+                    $e->getMessage(),
+                    PHP_EOL . $e->getTraceAsString()
+                )
+            );
+
+            $this->documentManager->flush();
+            $this->documentManager->clear();
+        }
+
+        return false;
+    }
+
+    /**
+     * @param BasePageDocument $document
+     * @param string $structureType
+     * @param string $webspaceKey
+     * @param string $locale
+     * @param string $format
+     * @param array $data
+     */
+    protected function setDocumentData(
+        BasePageDocument $document,
+        $structureType,
+        $webspaceKey,
+        $locale,
+        $format,
+        $data
+    ) {
+        $structure = $this->structureManager->getStructure($structureType);
+        $properties = $structure->getProperties(true);
+        $node = $this->documentRegistry->getNodeForDocument($document);
+        $node->setProperty(sprintf('i18n:%s-template', $locale), $structureType);
+        $state = $this->getParser($format)->getPropertyData('state', $data, null, null, 2);
+        $node->setProperty(sprintf('i18n:%s-state', $locale), $state);
+
+        if ($this->getParser($format)->getPropertyData('title', $data) === '') {
+            throw new \Exception(
+                sprintf('Document(%s) has not set any title', $document->getUuid())
+            );
+
+            return;
+        }
+
+        // import all content data
+        foreach ($properties as $property) {
+            $value = $this->getParser($format)->getPropertyData(
+                $property->getName(),
+                $data,
+                $property->getContentTypeName()
+            );
+
+            // don't generate a new url when one exists
+            $doImport = true;
+            if ($property->getContentTypeName() == 'resource_locator') {
+                if (!$document->getResourceSegment()) {
+                    $parent = $document->getParent();
+
+                    if ($parent instanceof BasePageDocument) {
+                        $parentUuid = $parent->getUuid();
+                        $value = $this->generateUrl(
+                            $structure->getPropertiesByTagName('sulu.rlp.part'),
+                            $parentUuid,
+                            $webspaceKey,
+                            $locale,
+                            $format,
+                            $data
+                        );
+                    }
+                } else {
```

Comment:
dont use `else` here you can set this var before the if and re-set it to true if you enter the if (or change the logic)